### DISCUSSION TITLE
Next.js Serverless Function's pages exceed max function size due to 90MB of dependencies when deployed to Vercel

### DISCUSSION THREAD

**Discussion by rosnovsky** [vercel/vercel#6366] [Help] [answered]
Hey folks, 

I've got a strange situation here. I've got a Next.js 11 website with a blog. I've just upgraded all of my blog post images to use the new `<Image />` component with `StaticImport` (and blur placeholder!). 

Now, whenever I attempt to deploy to Vercel, I get errors like this:

```
Warning: Max serverless function size of 50 MB compressed or 250 MB uncompressed reached
Serverless Function's page: blog.js
Large Dependencies                                           Uncompressed size  Compressed size
.next/server/chunks                                                            91.2 MB            90 MB
node_modules/esbuild/bin                                                 7.86 MB           3.1 MB
node_modules/refractor/lang                                              719 kB           284 kB
node_modules/@mdx-js/mdx                                              531 kB           155 kB
node_modules/remark-mdx/node_modules                      436 kB           122 kB
All dependencies                                                                 106 MB          95.1 MB
```

The problem, I think is that now I actually import images instead of providing the `Image` component with an `src` string. With this being said, I can't wrap my head around how to decrease serverless function's dependencies here. Next.js' build output looks fine:

```
17:01:21.819 | Page                                             Size           First Load JS
17:01:21.819 | ┌ ○ /                                            3.84 kB           95.6 kB
17:01:21.819 | ├   /_app                                             0 B              83 kB
17:01:21.819 | ├ ○ /_offline                                  252 B           83.2 kB
17:01:21.819 | ├ ○ /404                                         3.33 kB      90.9 kB
17:01:21.819 | ├ ○ /about                                       1.87 kB      93.6 kB
[...]
17:01:21.819 | ├ λ /api/algoliasearch                           0 B          83 kB
17:01:21.819 | ├ λ /api/auth/[...auth0]                         0 B           83 kB
17:01:21.820 | ├ ● /blog                                        24.2 kB         116 kB
17:01:21.820 | ├ ● /blog/[slug]                                 12.3 kB      104 kB
17:01:21.820 | ├   ├ /blog/2010-2020
17:01:21.820 | ├   └ [+46 more paths]
17:01:21.820 | + First Load JS shared by all                    83 kB
[...]
```

Local `vc dev` or `yarn build` don't produce any errors, but as soon as I deploy, Vercel starts yelling at me :( Any ideas why I'm getting this and how to fix it?

**Comment by rosnovsky** [comment]
Could be related to https://github.com/vercel/next.js/issues/26297

**Comment by rosnovsky** [comment]
All goes well, builds look good.
<img width="805" alt="CleanShot 2021-06-22 at 15 39 47@2x" src="https://user-images.githubusercontent.com/2508576/123009912-7d7cde80-d372-11eb-8124-3cd7b5ff7fc5.png">

Yet the `server` folder includes all 90+ Mb of images that I've got, and these images somehow end up being dependencies for my tiny serverless functions. I'm so very lost :(

**Comment by grod220** [comment]
Having the same issue. Sadly have to revert to Nextjs 10 until it's resolved.

**Reply by NoahBres** [reply]
Also running into this issue. Do you have to revert to Next 10? Can't you just revert back to string image sources rather than static imports?

Unfortunately we don't get the dynamic blur placeholders but I am thinking about replacing it with the [next shimmer placeholder sample](https://github.com/vercel/next.js/blob/canary/examples/image-component/pages/shimmer.js) until the issue is resolved

**Reply by NoahBres** [reply]
^^^ This works by the way. Just deployed fine on our site. No downgrade necessary

**Reply by grod220** [reply]
Can you detail the difference between the string image source vs static import? In my case, it's a string url.

**Reply by NoahBres** [reply]
```javascript
import Picture from '../public/image/picture.png';

<Image src={Picture}/>

// vs

<Image src="/image/picture.png" />
```

The first one is required for the dynamic blur placeholders in Next 11 but gets included in the serverless function size which causes the massive sizes. The second works fine.

**Comment by leerob** [comment, marked as answer]
Replaced by https://github.com/vercel/vercel/discussions/6516, please continue the discussion there 🙏
